### PR TITLE
Update RESTful Integration and RESTful Binary Sensor

### PR DESCRIPTION
**Pull request by lcnittl**
## Proposed change

This PR sorts the config entries of RESTful and RESTful Binary Sensor (like they already were in RESTful Sensor) and adds config entries that were missing in the RESTful Binary Sensor documentation (like `icon`, `picture`, `force_update`, etc.). It also resolves some typos in the RESTful documentation.

## Type of change

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [x] Removed stale or deprecated documentation.

## Additional information

- ~~Link to parent pull request in the codebase:~~
- ~~Link to parent pull request in the Brands repository:~~
- ~~This PR fixes or closes issue: fixes #~~

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/binary_sensor.rest.markdown [hunk 1]
```diff
@@ -66,17 +66,29 @@ sensor:
 {% endraw %}
 
 {% configuration %}
-resource:
-  description: The resource or endpoint that contains the value.
-  required: true
+authentication:
```
Comment (frenck) [inline source/_integrations/binary_sensor.rest.markdown:69]: These are available via the integation configuration, however, you are editing the platform configuration? (please note, we are trying to get rid of platform configurations, so even if it was possible I don't think we should advertise this).

**Review by frenck — CHANGES_REQUESTED** [review]
Comment.

File: source/_integrations/binary_sensor.rest.markdown (hunk 1 above)
Comment (lcnittl) [reply source/_integrations/binary_sensor.rest.markdown:69]: I see. The `resource` and `authentication` key were amongst the config entries already before - they have just been sorted alphabetically (like in RESTful Sensor).
All other entries were "synced" from the [RESTful Sensor](https://www.home-assistant.io/integrations/sensor.rest/) documentation. If they should not be listed here for RESTful Binary Sensor, then they should - for the sake of consistency - also be removed from the RESTful Sensor?

**Comment by lcnittl** [comment]
I have reverted the changes to RESTful Binary Sensor and have instead added a "legacy info" to both RESTful Binary Sensor and RESTful Sensor. I have also included the necessary sub-keys for `binary_sensor` and `sensor` in the configuration overview of the `RESTful` integration and have merged the examples from both platform sensors to the integration page (with updated configs of course).

This became slightly bigger as originally planned, but I think it helps to clean up some of the confusion and helps "deprecation" of the platform configurations.

**Comment by lcnittl** [comment]
Hi @frenck -- any chance that this gets merged in the (near) future? Or shall I close the PR?

**Comment by lcnittl** [comment]
@frenck So we let this slip into the void? 👎 Very sad!

**Pull request closed without merging**